package protobuf;

import "Declarations.proto";
import "Skeleton.proto";

message Triangle {
	required uint32 vert1 = 1;
	required uint32 vert2 = 2;
	required uint32 vert3 = 3;
}

message Mesh {
	required string material_path = 1;

	required VertexDeclaration vertices_format = 3;

	 // 4-byte aligned, except doubles which are 8-byte aligned
	 // each component, though, is packed
	 // e.g. a vector of 3 chars will take 4 bytes, not 12
	required bytes vertices_data = 4;

	required Joint skeleton = 5;

	repeated Triangle triangles = 6;
}